packages/react/src/components/table-hoc/TableWithFilter.tsx

```diff
@@ -65,7 +66,7 @@ export const tableWithFilter = (
                     id={this.props.id}
                     className="coveo-table-actions"
                     filterPlaceholder={filterPlaceholder || config.placeholder}
-                    isAutoFocus
+                    isAutoFocus={config.filter.isAutoFocus}

```

Comment:
Since the filter is optional you should use optional chaining here
```suggestion
                    isAutoFocus={config.filter?.isAutoFocus ?? true}
```